src/collections/componentSet.ts

```diff
@@ -360,7 +402,7 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
     >;
   }
 
-  public add(component: ComponentLike, asDeletion?: boolean): void {
+  public add(component: ComponentLike, deletionType?: DestructiveChangesType): void {
```

Comment:
replacing the `asDeletion` param is the breaking change, which can also be seen on other methods